python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -49,26 +53,17 @@ def __init__(
         self,
         dtype: plc.DataType,
         sub_expr: Expr,
-        *children: Expr,
+        *children: FusedExpr,
     ):
         self.dtype = dtype
         self.sub_expr = sub_expr
+        assert all(isinstance(c, FusedExpr) for c in children)
```

Comment:
FWIW, if *we're* the only ones constructing `FusedExpr` in cudf-polars, then we should be able to rely on `mypy` to verify that this is the case.